components/dashboard/src/components/SelectIDEComponent.tsx

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { IDEOption, IDEOptions } from "@gitpod/gitpod-protocol/lib/ide-protocol";
+import { useEffect, useMemo, useState } from "react";
+import { getGitpodService } from "../service/service";
+import { DropDown2, DropDown2Element } from "./DropDown2";
+import Editor from "../icons/Editor.svg";
+
+interface SelectIDEComponentProps {
+    selectedIdeOption?: string;
+    useLatest?: boolean;
+    onSelectionChange: (ide: string, latest: boolean) => void;
+}
+
+export default function SelectIDEComponent(props: SelectIDEComponentProps) {
+    const [ideOptions, setIdeOptions] = useState<IDEOptions>();
+    useEffect(() => {
+        getGitpodService().server.getIDEOptions().then(setIdeOptions);
+    }, []);
+    const getElements = useMemo(() => {
+        return (search: string) => {
+            if (!ideOptions) {
+                return [];
+            }
+            const options = IDEOptions.asArray(ideOptions);
+            const result: DropDown2Element[] = [];
+            for (const ide of options.filter(
+                (ide) =>
+                    `${ide.label}${ide.title}${ide.notes}${ide.id}`.toLowerCase().indexOf(search.toLowerCase()) !== -1,
```

Comment:
nit: we can maybe simplify this: [[MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes)]
```suggestion
                    `${ide.label}${ide.title}${ide.notes}${ide.id}`.toLowerCase().includes(search.toLowerCase()),
```